pkg/cmd/run.go

```diff
@@ -144,30 +163,70 @@ func NewRunCommand(factory factory.Factory, streams genericclioptions.IOStreams)
 
 // Validate validates the arguments and flags populating RunOptions accordingly.
 func (o *RunOptions) Validate(cmd *cobra.Command, args []string) error {
+	// Selector can only be used in conjunction with tracer.
+	o.tracerDefined = cmd.Flag("tracer").Changed
+	if !o.tracerDefined && cmd.Flag("selector").Changed {
+		return fmt.Errorf(tracerNeededForSelectorErrString)
+	}
+
+	switch o.tracer {
+	case bpftrace, bcc:
+	default:
+		return fmt.Errorf(tracerNotFound, o.tracer)
+	}
+
+	parsed, err := tracejob.NewSelector(o.selector)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+	o.parsedSelector = parsed
+
+	// All filtering must be via selector if tracer is specified.
 	containerFlagDefined := cmd.Flag("container").Changed
-	switch len(args) {
-	case 1:
-		o.resourceArg = args[0]
-		break
-	// 2nd argument interpreted as container when provided
-	case 2:
-		o.resourceArg = args[0]
-		o.container = args[1]
-		if containerFlagDefined {
-			return fmt.Errorf(containerAsArgOrFlagErrString)
+	if o.tracerDefined && containerFlagDefined {
```

Comment:
I'm a bit lost as to what is happening here, this triggers when we do something like `--tracer=bpftrace` with `-c CONTAINER` ?

Reply:
Yep, this is to make sure old syntax `-c` and positional args are not mixed with new syntax `--tracer=... --selector=...`

Still a bit apprehensive about keeping both and part of me thinks that once we cut a new release we could maybe just do a breaking change and move over to new syntax. That would make it easier for user to understand semantics long term. WDYT?